### PR TITLE
fix(network): extra info can arrive before will send request

### PR DESCRIPTION
**Pull request by pavelfeldman**

### CODE REVIEW

File: src/client/network.ts [hunk 1]
```diff
@@ -421,13 +422,8 @@ export class Response extends ChannelOwner<channels.ResponseChannel, channels.Re
     return { ...this._headers };
   }
 
-  async finished(): Promise<Error | null> {
-    return this._wrapApiCall(async (channel: channels.ResponseChannel) => {
-      const result = await channel.finished();
-      if (result.error)
-        return new Error(result.error);
```
Comment (dgozman) [inline src/client/network.ts:428]: I don't see where we reject `this._finishedPromise`. Should we throw from somewhere? Perhaps we miss a test?

File: src/server/chromium/crNetworkManager.ts [hunk 2]
```diff
@@ -620,9 +629,7 @@ class ResponseExtraInfoTracker {
   }
 
   responseReceivedExtraInfo(event: Protocol.Network.responseReceivedExtraInfoPayload) {
-    const info = this._requests.get(event.requestId);
-    if (!info)
-      return;
+    const info = this._getOrCreateEntry(event.requestId);
```
Comment (dgozman) [inline src/server/chromium/crNetworkManager.ts:632]: What if we called `this._stopTracking()` for this info already, and then received extra info for the 10th redirect? It seems like we'll create a new info that will never clean up?

**Review by dgozman — APPROVED** [review]

File: src/server/chromium/crNetworkManager.ts (hunk 2 above)
Comment (pavelfeldman) [reply src/server/chromium/crNetworkManager.ts:632]: That is the case. It is unlikely though and I don't know how to mitigate it...

File: src/client/network.ts (hunk 1 above)
Comment (pavelfeldman) [reply src/client/network.ts:428]: We did not and we should not...

**Pull request merged**